core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2454,7 +2456,7 @@ private Scope createScopeForView(
             }
 
             Query query = parseView(originalSql, name, table);
-            analysis.registerTableForView(table);
+            analysis.registerTableForView(table, name, isMaterializedView);
```

Comment:
I don't quite follow the question. Let me try to explain my understanding here and see if it answers your concern; otherwise can you elaborate?

So for materialized views we have two possibilities: we either read from the existing storage table, or we query from the base table using the view SQL. In the latter case, a materialized view is treated the same as a regular view, and we track the view SQL as well as the view reference chain (see `testReferencedTablesWithMaterializedViews`). In the former case, currently `getMetadata().getTables()` will include only a reference to the view name, without including any reference to the storage table. I think this is a miss, and we should include a reference to the storage table as well (with the view in its `referenceChain`), but it's separate from this PR, IMO.

Reply:
Thank you !! It answered my question.